docs/Concepts/ArchitectureOverview.md

```diff
@@ -0,0 +1,11 @@
+---
+description: Teku architecture
+---
+
+# Teku architecture
+
+The following diagram outlines the Teku high-level architecture.
+
+![Architecture](../images/Architecture.png)
```

Comment:
One more nit:  in the "Beacon Node" section, I'd move the "Sync" box to the bottom, so from top to bottom it reads: "Sync", "P2P", "Spec"